test/factory/ProxyFactory.spec.ts

```diff
@@ -46,28 +47,30 @@ describe("ProxyFactory", async () => {
         }
     })
 
-    describe("createProxy", async () => {
+    describe("createProxyWithNonce", async () => {
+
+        const saltNonce = 42
 
-        it('should revert with invalid singleton address', async () => {
+        it('should revert if singleton address is not a contract', async () => {
             const { factory } = await setupTests()
             await expect(
-                factory.createProxy(AddressZero, "0x")
-            ).to.be.revertedWith("Invalid singleton address provided")
+                factory.createProxyWithNonce(AddressZero, "0x", saltNonce)
```

Comment:
Could we use another address here to avoid that the test conflicts with an address 0 check